### DISCUSSION TITLE
A new question

### DISCUSSION THREAD

**Discussion by Chan-ZhiHao** [neutrinolabs/xrdp#2174] [Q&A]
Sir,I have a question.As we know, RDP is a  multi-channel protocol。For example, I use clipboard or disk channel to copy a file from my desktop to a remote server，could I get the file and save it by xrdp？In other words， could I get the file by xrdp agent? Because I use xrdp for operation and maintenace audit  agent which can save the user message during one operation.
Thanks.

**Comment by matt335672** [comment]
Hi @Chan-ZhiHao 

I don't understand your question exactly. xrdp supports file operations within a desktop session. Are you trying to do something other than this?

**Comment by Chan-ZhiHao** [comment]
Hello sir. We can copy local files through clipboard channels to remote terminals. As an agent, can xrdp save the completed file  when parsing or encapsulating RDP protocol. For example, I send a TXT file to the remote terminal through the xrdp clipboard, and then xrdp saves the txt file on the local machine. I can open it and read it.Thanks.

**Reply by matt335672** [reply]
Thanks for the clarification.

The clipboard supports copying and pasting of text files using the desktop you are on. This works both ways. xrdp won't allow the file copy to be intercepted and saved somwhere else at the same time.

Does that answer your question?
